### PR TITLE
Improve configuration and DLQ masking

### PR DESCRIPTION
**Pull request by CodeDrivenMitch**
This PR improves configuration by introducing a builder. The ConfigurerModule has been changed to java to easily decompile for users in the IDE and docs have been added.

Three modes for the DLQ have been added. Please see the documentation on the DlqMode enum for more information.

### CODE REVIEW

**Review by smcvb — APPROVED** [review]
Just a couple of minor suggestions. In general, I think this looks good.

File: console-framework-client-spring-boot-starter/src/main/java/io/axoniq/console/framework/starter/AxoniqConsoleSpringProperties.java [hunk 1]
```diff
@@ -8,10 +9,13 @@ public class AxoniqConsoleSpringProperties {
     private Integer port = 7000;
     private String credentials = null;
     private String applicationName = null;
-    private boolean dlqEnabled = true;
+    private AxoniqConsoleDlqMode dlqMode = AxoniqConsoleDlqMode.FULL;
     private boolean secure = true;
     private Long initialDelay = 0L;
 
+    /**
+     * The host to connect to. Defaults to {@code framework.console.axoniq.io}.
```
[the remark this answers is not in the document]
Comment (CodeDrivenMitch) [reply console-framework-client-spring-boot-starter/src/main/java/io/axoniq/console/framework/starter/AxoniqConsoleSpringProperties.java:17]: I will change it to align with it.

File: console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleAggregateConfigurerModule.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import io.axoniq.console.framework.messaging.AxoniqConsoleWrappedEventStore
+import org.axonframework.common.ReflectionUtils
+import org.axonframework.config.AggregateConfiguration
+import org.axonframework.config.Configurer
+import org.axonframework.config.ConfigurerModule
+import org.axonframework.eventsourcing.EventSourcingRepository
+import org.axonframework.eventsourcing.eventstore.EventStore
+import org.axonframework.modelling.command.Repository
+
+class AxoniqConsoleAggregateConfigurerModule : ConfigurerModule {
```
[the remark this answers is not in the document]
Comment (CodeDrivenMitch) [reply console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleAggregateConfigurerModule.kt:28]: Yes. I wanted to move the ConfigurerModule to java as much as possible. It's not possible for this part without a bigger rewrite, so I encapsulated this into its own module.

File: console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleConfigurerModule.java [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+     * @return The builder with which you can further configure this module
+     */
+    public static Builder create(String environmentId, String accessToken, String applicationName) {
+        return new Builder(environmentId, accessToken, applicationName);
+    }
+
+    @Override
+    public void configureModule(@NotNull Configurer configurer) {
+        if (applicationName.trim().isEmpty()) {
+            throw new IllegalArgumentException("Application Name of AxonIQ Console Properties cannot be empty");
```
[the remark this answers is not in the document]
Comment (CodeDrivenMitch) [reply console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleConfigurerModule.java:75]: You're right!

File: console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleConfigurerModule.java [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    /**
+     * Creates the base builder with the required parameters. Defaults to the public production environment of AxonIQ
+     * console.
+     *
+     * @param environmentId   The environment identifier of AxonIQ Console to connect.
+     * @param accessToken     The access token needed to authenticate to the environment.
+     * @param applicationName The dispaly name of the application.
+     * @return The builder with which you can further configure this module
+     */
+    public static Builder create(String environmentId, String accessToken, String applicationName) {
```
[the remark this answers is not in the document]
Comment (CodeDrivenMitch) [reply console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleConfigurerModule.java:68]: I was on the fence. Builder, for me, implies a no-args constructor. But I wanted these fields to always be set. I'll change it to builder

File: console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleConfigurerModule.java [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+
+    /**
+     * Creates the {@link AxoniqConsoleConfigurerModule} with the given {@code builder}
+     *
+     * @param builder The configured builder for the {@link AxoniqConsoleConfigurerModule}
+     */
+    protected AxoniqConsoleConfigurerModule(Builder builder) {
+        this.environmentId = builder.environmentId;
+        this.accessToken = builder.accessToken;
+        this.applicationName = builder.applicationName.replaceAll("([\\[\\]])", "-");
```
[the remark this answers is not in the document]
Comment (CodeDrivenMitch) [reply console-framework-client/src/main/java/io/axoniq/console/framework/AxoniqConsoleConfigurerModule.java:49]: Done in various places.

**Pull request merged**